### PR TITLE
Add deleteAllAnnotations method to the sdk

### PR DESCRIPTION
**Pull request by sbazzaz**
Add deleteAllAnnotations method to the flutter sdk

## Support Ticket
https://support.pdftron.com/a/tickets/21782

## Screenshots

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -362,6 +362,15 @@ list.add(new Annot('World', 2));
 PdftronFlutter.deleteAnnotations(annotList);
 ```
 
+#### deleteAllAnnotations
+Deletes all annotations in the current document.
```
Comment (sgong-pdftron) [inline API.md:366]: add "Exclude links and widget"

File: android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java [hunk 2]
```diff
@@ -33,9 +33,11 @@
 import com.pdftron.pdf.tools.QuickMenuItem;
 import com.pdftron.pdf.tools.Tool;
 import com.pdftron.pdf.tools.ToolManager;
+import com.pdftron.pdf.utils.AnalyticsHandlerAdapter;
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java:36]: remove

File: android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java [hunk 3]
```diff
@@ -2417,6 +2425,41 @@ private static void closeAllTabs(MethodChannel.Result result, ViewerComponent co
...
+            // Locks the document first as accessing annotation/doc information isn't thread
+            // safe.
+            pdfViewCtrl.docLock(true);
+            shouldUnlock = true;
+
+            AnnotUtils.safeDeleteAllAnnots(pdfDoc);
+            pdfViewCtrl.update(true);
+            hasChange = pdfDoc.hasChangesSinceSnapshot();
+        } catch (Exception e) {
+            AnalyticsHandlerAdapter.getInstance().sendException(e);
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java:2448]: just print out e.printStackTrace();

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

File: API.md (hunk 1 above)
Comment (sbazzaz) [reply API.md]: Done.

File: android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java (hunk 2 above)
Comment (sbazzaz) [reply android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java]: Done.

File: android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java (hunk 3 above)
Comment (sbazzaz) [reply android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java]: Done.

**Review by sgong-pdftron — APPROVED** [review]

File: ios/Classes/PdftronFlutterPlugin.m [hunk 4]
```diff
@@ -1548,6 +1550,64 @@ - (void)deleteAnnotations:(NSString *)annotationList resultToken:(FlutterResult)
...
+    } error:&error];
+
+    if (error) {
+        NSLog(@"Error: Failed to delete all annotations from doc. %@", error.localizedDescription);
+
+        flutterResult([FlutterError errorWithCode:@"delete_all_annotations" message:@"Failed to delete annotations" details:@"Error: Failed to delete annotations from doc."]);
+        return;
+    }
+    [documentController.pdfViewCtrl Update:YES];
+    [documentController.toolManager changeTool:[PTPanTool class]];
```
Comment (JamesTron) [inline ios/Classes/PdftronFlutterPlugin.m:1606]: We should call these even when there is an error.

**Pull request merged**